### DISCUSSION TITLE
Get basins id used in previous run

### DISCUSSION THREAD

**Discussion by issamsouassi** [neuralhydrology/neuralhydrology#182] [Q&A] [answered]
Hi all,
I hope you're doing well.

While doing the Multi-basins training, I was wondering if there is any way to get the ids of basins used for train/validation/test in a previous run ? I took a look in run directory I didn't find anything related to basins used (sometimes I modify the text file of basins id and I need to know what basins I did use for a previous run.

Thank you.

**Comment by gauchm** [comment, marked as answer]
For validation/test it's probably clear -- the results netcdf will show you what basins were used in inference.

For training, there's no explicit way that this would be stored. Only the path to the basin files is stored, but not the basins themselves. The only other thing that might help you in identifying the list is that you'll find an entry "number_of_basins" in the config.yaml in the run directory, which contains the number of training basins. If you really (_really_) need to find out, you can also brute force it by comparing the scaler values that result from different basin sets :)

**Comment by darylsuen** [comment]
As gauchm says, just open the path to the train_basin_file, validation_basin_file, and test_basin_file in the config file and the basin ids will be in those txt files. You could start adding a comment to each config file or, better, include the datetime in the .yml config filename and that way you can easily find those txt files for any desired run.
Regards.

**Comment by issamsouassi** [comment]
This was so helpful. Thanks a lot @gauchm, @darylsuen.
